### ISSUE TITLE
Adventure "Arctic Explosion"

### ISSUE THREAD

**Issue by MadMNN** [perceptron8/pinky.ext#63]
please add artifacts from the adventure  "Arctic Explosion" 
thank you very much.

**Comment by MadMNN** [comment]
Do not leave the problem unattended. this adventure is part of Halloween 2019.

**Comment by perceptron8** [comment]
It should be fixed in 94733c10bb5ec3a8542423751f67345294d5f151. Please wait for v0.4.8 and let me know if all pictures were added.

**Comment by MadMNN** [comment]
http://prntscr.com/pjnvlt
not highlighted

**Comment by perceptron8** [comment]
Can you tell me their names?

**Comment by MadMNN** [comment]
how can i parse a picture? 
only in russian: Тележка, Ящики

**Comment by perceptron8** [comment]
I was asking not about filenames, but rather collectible names - and russian is just as fine as any other language.

I've managed to find them already: [81e7f9a675419291fe18e576bb2be422b7614e67.png](https://ubistatic-a.akamaihd.net/0018/live/GFX_HASHED/building_lib/81e7f9a675419291fe18e576bb2be422b7614e67.png) and [70f3396b618644a2b525be39c8da046681ad59cc.png](https://ubistatic-a.akamaihd.net/0018/live/GFX_HASHED/building_lib/70f3396b618644a2b525be39c8da046681ad59cc.png).

**Comment by MadMNN** [comment]
yes they are but they are there white

**Comment by perceptron8** [comment]
They are respectively: [o_barrel_transporter](https://ubistatic-a.akamaihd.net/0018/live/GFX_HASHED/building_lib/81e7f9a675419291fe18e576bb2be422b7614e67.png) and [o_crates](https://ubistatic-a.akamaihd.net/0018/live/GFX_HASHED/building_lib/70f3396b618644a2b525be39c8da046681ad59cc.png).

They are white because of some graphic filter post processing.

**Comment by MadMNN** [comment]
these drawings are not only in this adventure in the game.

**Comment by perceptron8** [comment]
Yeah, that's what I was afraid of. See https://github.com/perceptron8/pinky.ext/issues/47. It's the same, but reversed. In Besieged City, collectible is used as decorative item. Here, decorative item is used as collectible. I'm not sure if I can do something here. It may be impossible to fix this and don't break something else.

**Comment by MadMNN** [comment]
then I download the script locally and put it in developer mode. thank you very much for your time

**Comment by perceptron8** [comment]
You're welcome! I wish I can do something more, but can't really. Hope you understand.
I'll keep this issue open as a reminder in case solution becomes possible one day.

**Comment by MadMNN** [comment]
I would be grateful if you taught to parse pictures

**Comment by perceptron8** [comment]
@MadMNN Or did you meant that you wish Pinky was implemented using [CV](https://en.wikipedia.org/wiki/Computer_vision) algorithms?

**Comment by MadMNN** [comment]
computer vision is definitely not that. the answer about the network came up
:)

**Comment by MadMNN** [comment]
as an option, save the picture locally on disk and search for it on the screen and replace it.
but this is a completely different implementation

**Comment by perceptron8** [comment]
Oh, that would require completely different set of permissions too. Instead I could let users choose what images should be replaced with pink, but for the sake of simplicity and ease of use I won't ever do that. It wouldn't be user friendly at all.

BTW I'd rather intercept and modify /GameServerXY/amf requests and automate collectibles picking entirely. Currently it's almost impossible with https://developer.chrome.com/extensions/webRequest and not so easy with https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/webRequest.

**Comment by bartalivini** [comment]
would it be possible to allow the user to "toggle" which collectibles are highlighted?
maybe simplifying with just 1 switch for the problematic ones used both as collectibles and decoration.
I'd rather allow some graphic glitches somewhere else but being able to spot all the collectibles during event rushes when one just repeats the same adventure over and over. switching it off then outside of events.

**Comment by perceptron8** [comment]
@bartalivini It's easier said than done. Preserving toggles between updates requires dealing with [identity](https://en.wikipedia.org/wiki/Identity_(philosophy)) of collectibles (so, at least, giving them unique names). It would be nice to show some labels too, not just images - that implies introduction and maintenance of [i18n](https://developer.chrome.com/apps/i18n). Pinky doesn't have any UI so far and I wish it could stay that way. Most importantly, most users know nothing about caching and I don't expect them to understand why there's no instant feedback when they toggle ("I just did it and nothing happened"). My current opinion is that it's not something worth doing, but I can't exclude that I change my mind it in the future. That's all I can promise, sorry.

**Comment by bartalivini** [comment]
@perceptron8 The extension is super useful even as it is. Arctic Explosion is one of the worst adventures eyesight-wise, it's bright white and collectibles are tiny and hidden, that's why I jumped in.

Thank you for your efforts (past, present and future) either way.

**Comment by vonhesperus** [comment]
Well, the items to find in arctic explosion are always at the same position, - so why the need to highlight them. Just make a map or look at one where they are shown..
